Doc style and link enhancements

michelle-purcell opened the pull request:
This PR fixes style issues (title capitalization, one sentence per line...), links, and some grammar as per the preferred Quarkus Contributor guidelines for writing tutorials.
This also harmonizes the content with product docs downstream for the 3.2 release.

michelle-purcell committed:
Doc style and link enhancements

fix typo

Fixes

Minor fixes

fix

fix

remove html file

Peer review fixes

Fixes to section headings level 3

fix

minortweak

Update docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

Co-authored-by: Sergey Beryozkin <sberyozkin@gmail.com>

Update docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

fix

sheilamjones commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -80,17 +82,19 @@ public class AdminResource {
-Injection of `JsonWebToken` is supported in `@ApplicationScoped`, `@Singleton` and `@RequestScoped` scopes however the use of `@RequestScoped` is required if the individual claims are injected as simple types, please see xref:security-jwt.adoc#supported-injection-scopes[Support Injection Scopes for JsonWebToken and Claims] for more details.
+Injection of `JsonWebToken` is supported in `@ApplicationScoped`, `@Singleton`, and `@RequestScoped` scopes.
+However, the use of `@RequestScoped` is required if the individual claims are injected as simple types.
+For more information, see the xref:security-jwt.adoc#supported-injection-scopes[Support Injection Scopes for JsonWebToken and Claims]  section of the Quarkus "Using JWT RBAC" guide.
```
Minor/non-urgent suggestion: Should the capitalization of the URL text be in sentence case.

michelle-purcell replied:
@sheilamjones  Fixed. Good catch. Thanks :100: 
_Your comment also triggered a review of the capitalization in some level 3 section headings. Fixed a few of those too._

sheilamjones commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -99,19 +103,20 @@ The default tenant's `OidcConfigurationMetadata` is injected if the endpoint is
 
-Additionally, a custom `SecurityIdentityAugmentor` can also be used to add the roles as documented in xref:security-customization.adoc#security-identity-customization[Security Identity Customization].
+A custom `SecurityIdentityAugmentor` can also be used to add the roles.
+For more information, see the xref:security-customization.adoc#security-identity-customization[Security Identity Customization] section of the Quarkus "Security tips and tricks" guide.
```
Minor/non-urgent: Capitalization of the URL text. Should this be "Security identity customization"

sheilamjones commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -154,27 +159,30 @@ public class ProtectedResource {
-If no matching `JWK` is available after the refresh then the JWT token will be sent to the OpenID Connect Provider's token introspection endpoint.
+If the token is a JWT token, then, by default, it is verified with a `JsonWebKey` (JWK) key from a local `JsonWebKeySet`, retrieved from the OIDC provider's JWK endpoint.
+The key identifier (`kid`) header value is used to find the matching JWK key.
+If no matching `JWK` is available locally, then `JsonWebKeySet` is refreshed by fetching the current key set from the JWK endpoint.
```
Suggestion: For clarity, propose to say "If no matching `JWK` key is available...."

michelle-purcell replied:
@sheilamjones - Fixed. Thanks!

sheilamjones commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -154,27 +159,30 @@ public class ProtectedResource {
+If no matching `JWK` is available locally, then `JsonWebKeySet` is refreshed by fetching the current key set from the JWK endpoint.
+The `JsonWebKeySet` refresh can be repeated only after the `quarkus.oidc.token.forced-jwk-refresh-interval` expires.
+The default expiry time is 10 minutes.
+If no matching `JWK` is available after the refresh, the JWT token is sent to the OIDC provider's token introspection endpoint.
```
Same comment as for line 169, for clarity, should we say "...`JWK` key..." here?

michelle-purcell replied:
@sheilamjones - +1. Thanks for flagging here and elsewhere. All instances are now fixed.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -970,44 +1040,50 @@ Note that `@TestSecurity` `user` and `roles` attributes are available as `TokenI
 ====
 
-=== How to check the errors in the logs
+=== Check log errors
```
```suggestion
=== Check errors in the log
```

sberyozkin replied:
Otherwise it reads as if we are checking `log errors` which may not be necessary related to OIDC. I have to admit I'm not sure what reads better `in the log` or `in the logs`...

michelle-purcell replied:
@sberyozkin - Makes sense. Thanks!

sberyozkin commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -393,12 +421,19 @@ Start by adding the following dependencies to your test project:
 
-Add the following dependencies to your test project:
 
+[[wiremock-integration-testing]]
```
I'm not sure the label name change is necessary, there might already be be external links pointing to it (from the code flow docs for example, from somewhere else), other labels follow a different pattern, it all can be realigned but it is not worth it IMHO :-)

michelle-purcell replied:
@sberyozkin - Good point. Fixed. I think I was trying to fix a broken anchor, and somehow I decided to recreate it this way :thinking:  I have reverted it back and kept the anchor fix in but with the original ID. Thanks!

sberyozkin commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -393,12 +421,19 @@ Start by adding the following dependencies to your test project:
 
-[[integration-testing-wiremock]]
-==== Wiremock
+//@sberyozkin - It might be good to add a stament here about the different mechanisms or methods to integration testing. For example, WireMock... etc
```
Sure, may be just underneath `Testing section` above - but I can't promise I'll remember this comment :-), may be worth opening an issue ? Or I can push an update to your PR ?

michelle-purcell replied:
@sberyozkin - :+1: +1 - An update to my PR would be perfect. Thank you.

sberyozkin replied:
Sounds good @michelle-purcell

sberyozkin commented:
Hi @michelle-purcell It all looks very good, significant improvements to the quality of the text, thanks. Thanks @sheilamjones too.

I've only opened a couple of suggestions - if you are OK with them then I can push the updates to your PR during the next day or two and merge it

michelle-purcell commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -970,44 +1040,50 @@ Note that `@TestSecurity` `user` and `roles` attributes are available as `TokenI
 ====
 
-=== How to check the errors in the logs
+=== Check errors in the log
```
```suggestion
=== Check errors in the logs
```

michelle-purcell commented:
@sberyozkin - Thanks so much for reviewing the PR before I head off for PTO :tada:  I have fixed all suggestions and rebased them. There's just that one outstanding intro bit left ^^ that you are going to add and merge into the PR, if git lets you. :green_heart: 

@sheilamjones - Thank you too, Sheila for reviewing :green_heart:

sberyozkin commented:
Looks like it was cancelled and I missed merging it.

sberyozkin commented:
It is upstream only though so let me merge it now, though I'll rebase

sberyozkin commented:
Hmm, `exec-maven-plugin:3.0.0:exec (assemble-downstream-doc) @ quarkus-documentation` is hanging, but I think it just finished on my laptop, I was checking Michelle's branch.

sberyozkin commented:
@gsmet I'm not sure why this PR is hanging, I don't see anything in the updated doc that could've caused it. Please review and merge if it looks fine. on PTO tomorrow

inoxx03 commented:
@sberyozkin Thasnk for the head's up. TBH, I have not found the cause yet.  trheI'm not sure but `assemble-downstream-doc` might be one of the post-processing jobs that we use to transform parts of thew upstream doc's AsciiDoc source to get around the downstream publishing system limitations.

 I'll come back and investigate further today, just have to finish some release-related tasks.

sheilamjones commented:
Apologies for slow reply @sberyozkin. Was tied up with 3.2 release tasks yesterday. Yes, as Stefan suggested, I think the assemble-downstream.doc topic relates to the downstreaming script with the post-processing fixes to enable us to bring the content downstream. Shall investigate further.

sberyozkin commented:
@sheilamjones No problems at all Sheila, thanks for the update.

Unfortunately it is still failing, something in this PR or in some other PRs is causing this task to fail.

@rsvoboda Rostislav, can you spot it ? Your fix is in Michelle's branch: https://github.com/michelle-purcell/quarkus/commits/security-doc-fixes-3-2-bearer-concept

rsvoboda commented on docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc:
```diff
@@ -393,12 +423,19 @@ Start by adding the following dependencies to your test project:
+You can also use link:https://wiremock.org/[WireMock] for integration testing.
 
+. Add the following dependencies to your test project:
++
+====
```
This noise makes the build of downstream docs to be stuck

there should be no `+` and `====` on solo lines

CC @sberyozkin @michelle-purcell

sberyozkin replied:
@rsvoboda @michelle-purcell It took me may be 5 rounds to get past the downstream task, I've ended removing all of standalone `+` and `====` and it looked like it fixed. Michelle, not sure if te end result would be what you expected it to be without these characters, so I guess you should have a look before me merging this PR

Thanks

sberyozkin commented:
Michelle, I'm not sure the numbering scheme works, there are a few places with `1.` where another `.1` follows in the same section after one paragraph etc. Can we consider removing it ?

michelle-purcell commented:
Thanks @sberyozkin. Apologies for not getting to this sooner. Between PTO and other priorities... 
Some of the numbering markup edits were introduced to fix bad numbering/style on the downstream publishing output too and to follow the advice in the upstream tutorial/how-to templates. 
I will test and investigate today and test the output of the markup on both publishing platforms. Thanks for flagging :-)

sberyozkin commented:
Hi Michelle @michelle-purcell , missed your comment, sorry. Sure please check it

michelle-purcell commented:
Closing this PR as it has languished and overlaps with other changes. @rolfedh is taking over the edits for this guide in a fresh PR. Thanks @rolfedh. Please see the comments above for your reference. Thanks.
